### PR TITLE
unifying tests to use the same TEST_NAMESPACE var

### PR DESCRIPTION
**Pull request by ewolinetz**

### CODE REVIEW

**Comment by ewolinetz** [comment]
/hold
@periklis this should address your cert gen issue for your Kibana tests

**Comment by periklis** [comment]
/hold

Something seems fishy with the E2E tests. I am still investigating, how it is possible that they pass although we get log entries like the following in the EO log:
```
time="2020-04-06T20:10:08Z" level=warning msg="Unable to perform synchronized flush: Post https://elasticsearch.e2e-test-10047.svc:9200/_flush/synced: x509: certificate is valid for localhost, elasticsearch, elasticsearch.cluster.local, elasticsearch.e2e-test-2977.svc, elasticsearch.e2e-test-2977.svc.cluster.local, not elasticsearch.e2e-test-10047.svc"
```

File: Makefile [hunk 1]
```diff
@@ -168,8 +168,10 @@ undeploy:
 #
 # You must also set:
 # ELASTICSEARCH_OPERATOR_NAMESPACE (Default: openshift-operators-redhat)
-test-e2e-olm: gen-example-certs
-	hack/test-e2e-olm.sh
+TEST_NAMESPACE="$${TEST_NAMESPACE:-e2e-test-$${RANDOM}}"
+test-e2e-olm: DEPLOYMENT_NAMESPACE="${TEST_NAMESPACE}"
+test-e2e-olm: gen-example-certs 
+	TEST_NAMESPACE=${TEST_NAMESPACE} hack/test-e2e-olm.sh
```
Comment (periklis) [inline Makefile:175]: The result will not be as expected with this syntax. First of all, the env var RANDOM will be read multiple times resulting in additional unique namespaces on each each read invocation, whether we want a single TEST_NAMESPACE. Furthermore, providing a Makefile override to TEST_NAMESPACE is simpler with `?=`. I propose to changes these lines to:
```
RANDOM_SUFFIX:=$(shell echo $$RANDOM)
TEST_NAMESPACE?="e2e-test-$(RANDOM_SUFFIX)"
test-e2e-olm: DEPLOYMENT_NAMESPACE=$(TEST_NAMESPACE)
test-e2e-olm: gen-example-certs
	TEST_NAMESPACE=$(TEST_NAMESPACE) hack/test-e2e-olm.sh
```

**Pull request merged**